src/com/google/enterprise/adaptor/database/UniqueKey.java

```diff
@@ -160,6 +161,57 @@
     return aclSqlCols;
   }
 
+  /**
+   * Attempt to extract ColumnTypes from the supplied Map
+   * of column names to java.sql.Type
+   *
+   * @param sqlTypes Map of column names to Integer java.sql.Types
+   */
+  void addColumnTypes(Map<String, Integer> sqlTypes) {
```

Comment:
Will the type map be the only optional builder parameter? Maybe additional ctor is enough? If there will be more optional parameters then maybe builder is the right idea.